### PR TITLE
Service Catalog Integration

### PR DESCRIPTION
**Pull request by mohitkumarsharmaflux7**
This PR creates Service Catalog product and adds it to  a given portfolio

### CODE REVIEW

File: cf-templates/common/microservice.yml [hunk 1]
```diff
@@ -6,13 +6,13 @@ Parameters:
     Description: Repository for getting the container
     Type: String
   ContainerName:
-    Description: Name for container used in various places
+    Description: Name  for container used in various places
```
Comment (sampritavh) [inline cf-templates/common/microservice.yml]: Why has these templates been modified? I don't think we should touch these templates

File: scripts/service-catalog.py [hunk 2]
```diff
@@ -0,0 +1,270 @@
...
+        #TO associate the product with portfolio
+        attach_product_to_portfolio(ser_cat_clt_conn,product_id,portfolio_id)
+        print "product {} attached with portfolio {}".format(product_name,PORTFOLIO_NAME)
+
+
+
+
+if __name__ == "__main__":
+    ARGS = parse_arguments()
+    SUPPORT_EMAIL = "mohit@flux7.com"
```
Comment (sampritavh) [inline scripts/service-catalog.py]: Do not hard-code these in the script. These should be taken as environment variables/cli args

File: scripts/service-catalog.py [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+            vdict= {}
+            for version in  version_response['ProvisioningArtifacts']:
+                tdict[time.mktime(version['CreatedTime'].timetuple())]=version['Id']
+                vdict[time.mktime(version['CreatedTime'].timetuple())]=version['Name']
+
+            latest_version_id= tdict[max(tdict.keys())]
+            latest_version_name= vdict[max(vdict.keys())]
+            print latest_version_id
+            print latest_version_name
+            template_utl = get_latest_version_template(ser_cat_clt_conn,latest_version_id,product_id)
```
Comment (sampritavh) [inline scripts/service-catalog.py]: template_utl??

File: scripts/service-catalog.py [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+            print "product_template=",product_template
+
+            comp_status = compare_templates(conn,template_utl,product_name,product_template)
+
+            # create version of product if template changed
+            if comp_status[0] == True:
+                global VERSION
+                VERSION = "v"+str(float(latest_version_name.split("v")[1])+.1)
```
Comment (sampritavh) [inline scripts/service-catalog.py]: so how does versioning scheme work? is it like v1.0 v1.1 ..? I was thinking we could do something like v1.0, v2.0 etc

**Comment by sampritavh** [comment]
We also need to change the template here: https://github.com/Flux7Labs/ecs-workshop-lz/blob/master/cf-templates/service/service-manager.yml#L136  to invoke the script to create/update service catalog product

**Comment by sampritavh** [comment]
[ECS-Workshop: Create Service Catalog products for microservice deployment pipeline](https://trello.com/c/7D3hZMEW/151-ecs-workshop-create-service-catalog-products-for-microservice-deployment-pipeline)

**Comment by sampritavh** [comment]
[ECS-Workshop: Create Service Catalog products for microservice deployment pipeline](https://trello.com/c/7D3hZMEW/151-ecs-workshop-create-service-catalog-products-for-microservice-deployment-pipeline)

File: cf-templates/common/microservice.yml (hunk 1 above)
Comment (mohitkumarsharmaflux7) [reply cf-templates/common/microservice.yml]: i just modified the description  string to check the updation of version is working or not. i also removed the modification.

File: scripts/service-catalog.py (hunk 2 above)
Comment (mohitkumarsharmaflux7) [reply scripts/service-catalog.py]: yes i am going to change it

Comment (mohitkumarsharmaflux7) [reply scripts/service-catalog.py]: replace with cli args

File: scripts/service-catalog.py (hunk 4 above)
Comment (mohitkumarsharmaflux7) [reply scripts/service-catalog.py]: it is like v1.0 then v1.1

**Pull request merged**